src/gestures/use-tap-gesture.ts

```diff
@@ -183,152 +179,98 @@ export interface TapHandlers {
     whileTap?: string | TargetAndTransition
 }
 
-/**
- * Explanation:
- *
- * Currently there's a problem where props can change while a gesture is active. In
- * this hook, we have event handlers that are created in `useMemo`, which is sync.
- * The event handlers themselves are activated in `useEvent` with `useEffect`, which
- * is async.
- *
- * Most event handlers are being set like this: onTap={() => {}}
- * This creates a new function every render, thus breaking `useMemo`. When any props
- * are set during a gesture, this creates a new event. We then have a `useEffect` in this
- * hook that, when props change, unsets the previous event handlers. I suspect that at
- * some point, the pointer up event is fired in between the previous one
- * being deactivate and the new one being activated.
- *
- * I attempted to rewrite the hook without `useMemo` at all but encountered the same issue.
- * At some point the `pointerUp` event is being unset and not reapplied until after it fires.
- *
- * By setting props to a ref we maintain longer gesture sessions and don't encounter this funny
- * business, but I'd rather a solution that required less code.
- *
- * @param props
- */
-const usePropsRef = <T>(props: T) => {
-    const propsRef = useRef(props)
-
-    for (const key in props) {
-        propsRef.current[key] = props[key]
-    }
-
-    return propsRef.current
-}
+type RemoveEvent = () => void
 
 /**
  * @param handlers -
  * @internal
  */
 export function useTapGesture(
-    handlers: TapHandlers & ControlsProp
-): { onPointerDown: EventHandler }
-export function useTapGesture(
-    handlers: TapHandlers & ControlsProp,
+    {
+        onTap,
+        onTapStart,
+        onTapCancel,
+        whileTap,
+        controls,
+    }: TapHandlers & ControlsProp,
     ref: RefObject<Element>
-): undefined
-export function useTapGesture(
-    props: TapHandlers & ControlsProp,
-    ref?: RefObject<Element>
-): undefined | { onPointerDown: EventHandler } {
+) {
+    const hasTapListeners = onTap || onTapStart || onTapCancel || whileTap
     const isTapping = useRef(false)
-    const { onTap, onTapStart, onTapCancel, whileTap, controls } = props
-    const propsRef = usePropsRef(props)
+
+    const pointerEventSubscription = useRef<RemoveEvent | undefined | null>(
```

Comment:
This is not really a 'subscription' right? More like a `cancelPointerEventHandler`?